raft.go

```diff
@@ -1468,3 +1632,71 @@ func (r *Raft) setState(state RaftState) {
 		r.observe(state)
 	}
 }
+
+// LookupServer looks up a server by ServerID.
+func (r *Raft) lookupServer(id ServerID) *Server {
+	for _, server := range r.configurations.latest.Servers {
+		if server.ID != r.localID {
+			return &server
+		}
+	}
+	return nil
+}
+
+// pickServer returns the follower that is most up to date.
+func (r *Raft) pickServer() *Server {
+	target := r.getLastIndex()
+	var pick *Server
+	var current uint64
+	for _, server := range r.configurations.latest.Servers {
+		if server.ID == r.localID {
+			continue
+		}
+		state, ok := r.leaderState.replState[server.ID]
```

Comment:
Is this OK? In general leaderState is expected to only be accessed by the single goroutine running the leader loop. Does this always run there? I think it does but worth checking. If so I suggest we update the comments on these new methods that assume they are part of the leader loop to make that explicit?

Reply:
Great catch, it wasn't called from the leaderloop! Plus I found another place where `leaderstate` is accessed outside of the `leaderloop`. Fixed both things.

Reply:
https://github.com/hashicorp/raft/pull/306/commits/20a87dbeaa9522f3e5305dc1fe2c705a8ed22fbb